library/src/test/java/com/datatorrent/lib/filter/FilterTest.java

```diff
@@ -173,9 +175,18 @@ public void testFilterError()
 
     clearFilterOperator();
   }
+  
+  @Test
+  public void testOptionalExpressionFunctions()
+  {
+    filter.setOptionalExpressionFunctions(Arrays.asList(new String[] {"org.apache.commons.lang3.BooleanUtils.*"}));
+    prepareFilterOperator(DummyPublicPOJO.class, "({$}.val == 1)");
+    Assert.assertEquals(6, filter.getExpressionFunctions().size());
+  }
+  
```

Comment:
Can you also add test for setOptionalExpressionFunctionsItem?
Consider the out of bound condition where during createExpression, there will be some null elements in optionalExpressionFunctions.